test/topojson/test.ml

```diff
@@ -1 +1,54 @@
 (* Tests to validate implemented functions/ modules  *)
+
+let read_file f =
+  let ic = open_in f in
+  let rec loop acc =
+    try loop (input_line ic :: acc)
+    with End_of_file ->
+      close_in ic;
+      List.rev acc
+  in
+  let lines =
+    try loop []
+    with _ ->
+      close_in ic;
+      failwith "Something went wrong reading file"
+  in
+  String.concat "\n" lines
+
+module Ezjsonm_parser = struct
+  type t = Ezjsonm.value
+
+  let catch_err f v =
+    try Ok (f v) with Ezjsonm.Parse_error (_, s) -> Error (`Msg s)
+
+  let find = Ezjsonm.find_opt
+  let to_string t = catch_err Ezjsonm.get_string t
+  let string = Ezjsonm.string
+  let to_float t = catch_err Ezjsonm.get_float t
+  let float = Ezjsonm.float
+  let to_int t = catch_err Ezjsonm.get_int t
+  let int = Ezjsonm.int
+  let to_list f t = catch_err (Ezjsonm.get_list f) t
+  let list f t = Ezjsonm.list f t
+  let to_array f t = Result.map Array.of_list @@ to_list f t
+  let array f t = list f (Array.to_list t)
+  let to_obj t = catch_err Ezjsonm.get_dict t
+  let obj = Ezjsonm.dict
+  let null = `Null
+  let is_null = function `Null -> true | _ -> false
+end
+
+module Topojson = Topojson.Make (Ezjsonm_parser)
+
+let () =
+  let s = read_file "./test_cases/files/exemplar.json" in
+  let json = Ezjsonm.value_from_string s in
+  let topojson_obj = Topojson.of_json json in
+  match topojson_obj with
+  | Ok v -> (
+      match Topojson.to_json v with
+      | (Topology f,[]) -> (v,f)
```

Comment:
> what I am missing in this one, can we do like this (showing errors, missing something 😞 )

So like the previous error we're dealing with the typechecker again. We have in general two possibilities. Either the function is not returning the type we expected (so we need to change the function) or we are assuming the wrong type (and need to change the code where the error is). Here's the error:

```
File "test/topojson/test.ml", line 51, characters 8-23:
51 |       | (Topology f,[]) -> (v,f)
             ^^^^^^^^^^^^^^^
Error: This pattern matches values of type 'a * 'b
       but a pattern was expected which matches values of type Topojson.json
```

Here you have pattern-matched on a pair hence `'a * 'b`. Do we expect a pair? The actual type is `Topojson.json`, is this expected? You have first converted the `json` to a `t` with `Topojson.of_json json` and then you convert it back to the raw `json` with `Topojson.to_json v` so that's right. This means your pattern-match is incorrect.

Reply:
Will work on it to successfully match the pattern